### ISSUE TITLE
S3 folder listing fails when >1000 files

### ISSUE THREAD

**Issue by rbarriuso** [dreamfactorysoftware/dreamfactory#134]
Our DreamFactory v2.4.1 is integrated with Amazon S3 via an "AWS S3"-type service, configured with a certain S3 container (bucket).
In that bucket there's a folder (let's say `"my_folder"`) and inside that folder there's a growing number of image files.

The DF endpoint to list the files in the folder (`/api/v2/s3/my_folder`) has been working fine until the file count exceeded around 1000 images. Now the server response is:

```
<!DOCTYPE HTML PUBLIC "-//IETF//DTD HTML 2.0//EN">
<html><head>
<title>503 Service Unavailable</title>
</head><body>
<h1>Service Unavailable</h1>
<p>The server is temporarily unable to service your
request due to maintenance downtime or capacity
problems. Please try again later.</p>
<p>Additionally, a 503 Service Unavailable
error was encountered while trying to use an ErrorDocument to handle the request.</p>
</body></html>
```

And in the server logs the following message is printed:
```
[2016-12-27 16:13:45] local.ERROR: exception 'Symfony\Component\Debug\Exception\FatalErrorException' with message 'Maximum execution time of 500 seconds exceeded' in /opt/bitnami/apps/dreamfactory/htdocs/vendor/aws/aws-sdk-php/src/Api/AbstractModel.php:0
Stack trace:
#0 {main}  
```

We found [this topic in the forum](http://community.dreamfactory.com/t/fetch-more-than-1000-records-in-one-step) which may be related, though is not connected to AWS S3.

We read that [the list objects operation of the S3 REST APIis limited to 1000 records by default](http://docs.aws.amazon.com/AmazonS3/latest/API/v2-RESTBucketGET.html), but in any case I understand that the endpoint should at least return those 1000 records and a reference to continue fetching.

Is this a bug or a known limitation in DF?
Is there any way to workaround it?

Original forum post: http://community.dreamfactory.com/t/s3-folder-listing-fails-when-1000-files/3539 :

**Comment by drewpearce** [comment]
I filed this as DF-1115

**Comment by drewpearce** [comment]
Resolved in 2.8.0 release
